packages-oss.yml

```diff
@@ -0,0 +1,282 @@
+# packages.yml
+#
+# packages.yml defines all the packages we are able to build for a single commit
+# in this repo. A package means a single zip file containing the executable binary,
+# and optionally other files if needed.
+#
+# packages.yml is a convenience file for the human management of large numbers of
+# alternate packages, allowing default and templated values. We generate another
+# artifact from this one, called packages.lock which contains the fully expanded set
+# of package and layer specs. This fully expanded file is in turn used to drive the
+# build process, and as data for templating CI config.
+
+# config contains packagespec config for this repo.
+config:
+  # product-repo is important to CI providers.
+  product-repo:          git@github.com:hashicorp/vault.git
+  release-repo:          git@github.com:hashicorp/vault-release.git
+  # product-id is used by external systems to identify this product.
+  # It can be any unique name, but a golang import path is ideal.
+  product-id:            github.com/hashicorp/vault
+  circleci-project-slug: gh/hashicorp/vault
+  circleci-host:         circleci.com
+
+# inputs are a set of environment variables that may affect the set of bytes produced
+# for a given package. Note that a package is a zip file containing the binary, so
+# the name of the binary does affect the package's bytes.
+inputs:
+
+  # defaults contains default input values for each package.
+  # These values may be overridden on a per-package basis in the packages section.
+  defaults:
+
+    # PRODUCT_VERSION is the version of this product. Usually, this should be left
+    # as 0.0.0-snapshot. When we build a release candidate, this is overridden in
+    # a one-off fashion to produce that build.
+    # This should be used in the PACKAGE_NAME template.
+    PRODUCT_VERSION: 0.0.0-snapshot
+
+    # GO_VERSION is the version of the Go toolchain to use to compile this package.
+    GO_VERSION: 1.14.6
+
+    # YARN_VERSION is the version of Yarn to install for the UI layer.
+    YARN_VERSION: 1.19.1-1
+
+    # BUILD_TAGS are passed as-is to the -tag flag of 'go build'. It should be
+    # a comma or space separated list, depending on the GO_VERSION you are using.
+    BUILD_TAGS: vault
```

Comment:
I don't think comma separated was ever valid, as per https://go-review.googlesource.com/c/go/+/41951/.  Also, why are we specifying the build tag `vault`, which I've never seen before and doesn't seem to be referenced in any build pragmas?

Reply:
Comma separated `-tags` support was added recently (in support of making the the `GOFLAGS` env var easier to use). See `go help build` for a recent version of Go; comma-separated is now recommended, at least as of go1.13.12.

Reply:
Re the `vault` build tag, that came from here: https://github.com/hashicorp/vault/blob/master/scripts/build.sh#L17 although I'm sure it prob isn't necessary really.

Reply:
Yeah, it's not used now, if it ever was.  I'd remove it to avoid confusion.

Reply:
Done

Reply:
FYI this was backported to the release/1.2.x branch as well.